Add Pronatura Bydgoszcz

jestempablo opened the pull request:
Pronatura Bydgoszcz serves a part of Bydgoszcz, Poland

5ila5 commented:
Please provide a md file and run the updatedoculinks.py script like described in the [Source Contribution Guide](https://github.com/mampfes/hacs_waste_collection_schedule/blob/master/doc/contributing_source.md)

5ila5 commented on custom_components/waste_collection_schedule/waste_collection_schedule/source/pronatura_bydgoszcz.py:
```diff
@@ -0,0 +1,138 @@
+        self._lang = lang if lang in ["pl", "en"] else "pl"  # Fallback to 'pl' if not 'pl' or 'en'
+
+    def fetch(self):
+        try:
+            return self.get_data()
+        except Exception as e:
+            _LOGGER.error(f"fetch failed for source {TITLE}: {e}")
+            return []
```
Please do not catch any errors, they will be caught automatically

5ila5 commented on custom_components/waste_collection_schedule/waste_collection_schedule/source/pronatura_bydgoszcz.py:
```diff
@@ -0,0 +1,138 @@
+            for collection in month_schedule["schedule"]:
+                waste_type = collection["type"]
+                for day in collection["days"]:
+                    if waste_type in ["papier", "plastik", "szkło"]:
+                        recykling_days.add(int(day))
+                    else:
+                        entries.append(
+                            Collection(
+                                datetime.date(year, month, int(day)),
+                                name_map[waste_type],
+                                ICON_MAP[waste_type],
+                            )
+                        )
+            # Add recykling entries
+            for day in recykling_days:
+                entries.append(
+                    Collection(
+                        datetime.date(year, month, day),
+                        name_map["papier"],
+                        "mdi:recycle",
+                    )
+                )
```
Is there a reason you are mapping paper plastic and glass to recycling instead of showing all 3. Are they always at the same date? For all addresses?

jestempablo replied:
i will revert to separate because im not sure its all addresses, and that merge is not really necessary

5ila5 commented on custom_components/waste_collection_schedule/waste_collection_schedule/source/pronatura_bydgoszcz.py:
```diff
@@ -0,0 +1,138 @@
+
+TITLE = "Bydgoszcz Garbage Collection"
+DESCRIPTION = "Source for Bydgoszcz city garbage collection by Pronatura"
+URL = "https://zs5cv4ng75.execute-api.eu-central-1.amazonaws.com/prod"
```
URL is used by the updatedoculinks.py script and will be shown in the README.md use the website URL here instead and rename this variable to something like API_URL

jestempablo commented:
@5ila5 thanks for the comments! i updated PR now to remove unnecessary features (like lang and combined schedules) and updated the docs as per contribution guidelines

5ila5 commented:
Thanks for your contribution. Just did some minor clean up

- rename to match URL (add _pl)
- removed unnecessary method get_data
- NAME_MAP[waste_type] and ICON_MAP[waste_type] with get equivalents to prevent failour on name change
- modified md file to be more in line with other sources"
